src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -966,6 +966,22 @@ private function addMessengerSection(ArrayNodeDefinition $rootNode)
                 ->arrayNode('messenger')
                     ->info('Messenger configuration')
                     ->{!class_exists(FullStack::class) && class_exists(MessageBusInterface::class) ? 'canBeDisabled' : 'canBeEnabled'}()
+                    ->validate()
+                        ->ifTrue(function ($v) { return isset($v['adapter']) && isset($v['adapters']); })
+                        ->thenInvalid('"adapter" and "adapters" cannot be used together.')
+                    ->end()
+                    ->beforeNormalization()
+                        ->ifTrue(function ($v) {
+                            return is_array($v) && isset($v['adapter']);
+                        })
+                        ->then(function ($v) {
+                            $v['adapters'] = array('default' => $v['adapter']);
```

Comment:
Shouldn't it use the short syntax for arrays here?

Reply:
Yes! However, to keep consistency with all the existing PHP code, we decided to keep using the long `array()` syntax instead of `[...]` even for the new code added to the master branch (which requires PHP 7.1). It's shown here: https://symfony.com/doc/current/contributing/code/standards.html